### ISSUE TITLE
[wg/webextensions] Web Extensions Working Group

### ISSUE THREAD

**Issue by koalie** [w3c/strategy#146]
New charter proposal, reviewers please take note.

# Charter Review

[Charter](https://w3c.github.io/charter-drafts/2025/webextensions-wg.html)

[diff from charter template](https://services.w3.org/htmldiff?doc1=https://w3c.github.io/charter-drafts/charter-template.html&doc2=https://w3c.github.io/charter-drafts/2025/webextensions-wg.html)

Expected end of charter refinement phase: second half of September

[(CG) chair dashboard](https://www.w3.org/PM/Groups/chairboard.html?gid=cg/webextensions)

What kind of charter is this? Check the relevant box / remove irrelevant branches. 

* New
 - [x] New WG 

## Communities suggested for outreach

This came from the WebExtensions CG itself. Given the wide market for browser extensions, outreach could be large...

## Known or potential areas of concern

This will significantly increase our work area, in particular in horizontals: a11y, security, and privacy. Those experts are already stretched...

## Anything else we should think about as we review? 

cc @xeenon @dotproto

~~The Web Extensions draft spec has been developed in a W3C CG [Browser Extension CG](https://www.w3.org/community/browserext/
)  and is being implemented but not harmoniously, apparently.~~

~~Why isn't there a Working Group for this work and what is the REC track plan for Web Extensions? Or, what's the alternative to this spec?~~

**Comment by ianbjacobs** [comment]
Ian adds @plehegar to see what his thoughts are.

**Comment by dontcallmedom** [comment]
The recently launched [WebExtensions CG](https://www.w3.org/community/webextensions/) is taking over that topic, with interest and support from all major browser vendors. Moving to to incubation

**Comment by plehegar** [comment]
Sat down at TPAC 2024 with the CG folks. They need to write a draft specification before we can put a charter around it.

**Comment by plehegar** [comment]
Charter will be on the agenda of the [CG f2f next month](https://github.com/w3c/webextensions/issues/759).

**Comment by plehegar** [comment]
We received the draft from the CG. See [above for links](https://github.com/w3c/strategy/issues/146#issue-366720033).

**Comment by dotproto** [comment]
> Curious why there is no intention to take the specifications to Recommendation?

In short, we are currently pursuing approaching our specification as a living standard.

As I was looking over how other groups have approached their specifications and recent W3C guidance around deliverables for specifications -- especially @plehegar's TPAC 2020 [Living Standards and Reviews](https://www.youtube.com/watch?v=RnnwbPm1ijo) breakout ([slides](https://www.w3.org/2020/Talks/TPAC/W3C2020/?full#8)) -- my impression was that pursuing a Recommendation will be difficult because the webextensions platform is constantly evolving. If we started with the intention to pursue Recommendation, I fear we'd be setting ourselves up for a situation like WebRTC where we could be working on a spec for nearly a decade before they were able to reach Recommendation.

A living standard approach feels like the best fit for where we are right now, how our members have been collaborating in the WebExtensions Community Group (WECG), and how we foresee pursuing the specification in a potential working group. Given that desire, the model that seemed to best fit this was what PLH described as the "CR Forever" approach ([slide 8](https://www.w3.org/2020/Talks/TPAC/W3C2020/?full#8)).

Another major consideration was the possibility of changing our approach in the future. As I understand it, it is significantly easier to move from targeting Candidate Recommendation to Recommendation than it would be to try to transition into the opposite direction. 

> Why is this called "web extensions" when it is about _browser_ extensions?

This is largely an artifact of historical choices around the naming of previous development work and specification efforts in this space. When Mozilla decided to pursue an extension model that was compatible with the Chromium project's approach, they [chose the name](https://blog.mozilla.org/addons/2015/08/21/the-future-of-developing-firefox-add-ons/) "WebExtensions" to refer to their implementation. Unfortunately I don't have any additional insight into the considerations that went into the selection of the name at that time.

A few months before that, the Browser Extensions Community Group (BECG) formed aiming "to establish a set of standards for interoperable browser extensions" ([BECG site](https://browserext.github.io/)). The specification this group pursued was named [Browser Extensions](https://browserext.github.io/browserext/). Despite the commendable efforts of this group, my impression is this effort suffered from a lack of participation from browser vendors other than Mozilla.

Several years later, after Chrome began working on Manifest V3, multiple browser vendors began discussing the possibility of collaborating on a cross-browser extension development model. This in turn lead to the creation of the WECG. (Note: I helped facilitate the formation of this group on behalf of Chrome.) When forming this group, we wanted to use a name that would be broadly understood as referring to small 3rd party applications that augment user agents and to avoid collisions with per-existing groups and specs. I explicitly wanted to avoid obscuring the BECG's previous efforts or come across as trying to take over their work. As a result of Mozilla's work in this space, the terms "web extensions" and "webextensions" (often used interchangeably) were our most compelling options as they were established terms that developers recognized as referring to a cross-browser approach to extension development. We eventually aligned on the camel case version of the name in part because it more uniquely identified our specific effort.

As WECG members (namely myself and @oliverdunk) worked on this draft charter, we came across W3C guidance that recommended that we use the convention of "Web <TOPIC>" for either our group's name or specifications' name. (Unfortunately, I can't find the resource at the moment.) As a direct result, we decided to change the formal name to "Web Extensions Working Group" and to use "WebExtensions WG" as our short name.

I'm not aware of any previous W3C specifications that follow the convention of using "Browser <TOPIC>" or "User Agent <TOPIC>". That said, I think we'd be open to discussing this if there are broader concerns about the currently proposed name.

**Comment by plehegar** [comment]
[advance notice](https://lists.w3.org/Archives/Member/w3c-ac-members/2025JulSep/0028.html)

**Comment by plehegar** [comment]
Currently expected to go for AC Review in second half of September 2025.

**Comment by simoneonofri** [comment]
From the Security IG, we believe that standardization on the Web Extensions side is a particularly important element for the Threat Model for the Web. Therefore, we would like to ask you to consider adding the [Threat Model for the Web](https://github.com/w3c/threat-model-web), particularly for the part concerning Web Extensions.

**Comment by csarven** [comment]
The TAG acknowledges the long history of browser extensions and encourages their standardisation through the new WG.

The TAG recommends the following:

Incorporate a "Threat Modeling" document under the Other Deliverables section to help implementers and developers understand potential attack surfaces and mitigations.

Incorporate "authentication" under the Scope section to cover extension-specific credential handling, background access, and interactions with auth flows.

To ensure adequate coordination with other groups and organizations, the charter should include or clarify relevant coordination points. See issue https://github.com/w3c/charter-drafts/issues/690

As a side note, the charter could clarify whether sufficient incubation has occurred to navigate prior experience and developer feedback, such as lessons from Manifest V3.

**Comment by plehegar** [comment]
@xeenon @dotproto , would you like to create pull requests to address the comments from the TAG or proceed diffeerently ? Happy to attend one of the CG meetings if those items need discussion.

**Comment by npdoty** [comment]
No particular privacy concerns or blockers regarding the charter. However, we note that this is going to be very privacy-relevant work (adding surface area to the whole user agent itself, not just from a site), and will be challenging given the limitations of existing deployments. I recognize that 'app store' stuff is out of scope, but some things about reputation/integrity/audits/evaluations is going to be pretty important for a good cross-browser privacy and security plan.

**Comment by dotproto** [comment]
Thanks for the feedback, @csarven. I didn't see those updates until just now, so thanks also to @plehegar for checking in.

I'll start working on pull requests to address the issues identified in @csarven's [Sept 5th comment](https://github.com/w3c/strategy/issues/146#issuecomment-3257447876) and the suggestions in https://github.com/w3c/charter-drafts/issues/690.

**Comment by dotproto** [comment]
@npdoty, can you share more about what you mean when you say this "privacy-relevant work … will be challenging given the limitations of existing deployments"? I suspect that I'm probably just looking at this from the wrong angle.

At the moment the work I'm most focused on is specifying the common extension model that browsers already implement. From that perspective, I don't see what we're doing as "adding surface area to the whole user agent itself" as codifying the existing surface area that we all support. While we don't currently have an established process for raising, discussing, or working to resolve privacy and security concerns, each participating browser has had to grapple with these issues in their extension implementation over the years and as a result has established (at least some of) their own processes.

I'm very interested in your thoughts on what we can or should do differently. Might you be able to attend the next WECG metting on [Thursday, September 25 from 8-9 AM Pacific](https://www.w3.org/events/meetings/43e4fb8e-4736-445c-b051-2383ffeef033/20250925T080000/) to speak on this and the "reputation/integrity/audits/evaluations" idea you mentioned? If you're time constrained, I'd be happy to adjust the agenda to accommodate.

**Comment by dotproto** [comment]
Looking forward to working with you, @sideshowbarker!

**Comment by sideshowbarker** [comment]
Howdy @dotproto 
> I'll start working on pull requests to address the issues identified in [@csarven](https://github.com/csarven)'s [Sept 5th comment](https://github.com/w3c/strategy/issues/146#issuecomment-3257447876) and the suggestions in [w3c/charter-drafts#690](https://github.com/w3c/charter-drafts/issues/690).

If there’s anything I can do to help with those, please lemme know

**Comment by sideshowbarker** [comment]
> Rather than adding a section that specifically focuses on authentication, I think it may be better to:
> - add a scope entry stating that extensions are able to intermediate any browser operations that a user agent may choose to make available to extensions.
> - add a scope entry highlighting that extensions are able to modify network requests, which has implications on aspects of the web platform including, but not limited to, authentication, identity, navigation, and session management.

💯% agreed. That sounds to me like exactly what could be useful in the spec to accommodate what the TAG comment was designed to achieve. So in the interest of giving us all something concrete to review, I'd recommend you go ahead with creating a patch/PR which does exactly that

**Comment by npdoty** [comment]
@dotproto sorry, it was a busy time for me and I couldn't make that meeting. We didn't have concerns about the charter. I do think that documenting what browsers are already implementing and that extension developers are already using is an opportunity to improve privacy/security protection, but it's a new surface for us to discuss in the standards environment, and the feedback from the Privacy WG was that it might be hard to make a lot of changes because of those widely used but not yet standardized implementations.

I'm happy to chat with the CG sometime, or join a WG call after it's formed, just because it's definitely privacy-relevant work and better to think ahead about it as much as we can. But no blocking issue on the charter.

**Comment by plehegar** [comment]
[Proposed W3C Charter: WebExtensions Working Group (until 2026-04-27)](https://lists.w3.org/Archives/Public/public-new-work/2026Mar/0002.html)

**Comment by plehegar** [comment]
[Approved](https://lists.w3.org/Archives/Public/public-new-work/2026May/0003.html)
